model/project_ref.go

```diff
@@ -1226,13 +1231,39 @@ func (t *PatchTriggerDefinition) Validate(parentProject string) error {
 	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.PatchSucceeded, evergreen.PatchFailed}, t.Status) {
 		return errors.Errorf("invalid status: %s", t.Status)
 	}
-	_, regexErr := regexp.Compile(t.VariantRegex)
-	if regexErr != nil {
-		return errors.Wrap(regexErr, "invalid variant regex")
-	}
-	_, regexErr = regexp.Compile(t.TaskRegex)
-	if regexErr != nil {
-		return errors.Wrap(regexErr, "invalid task regex")
+
+	for _, specifier := range t.TaskSpecifiers {
+		if (specifier.VariantRegex != "" || specifier.TaskRegex != "") && specifier.PatchAlias != "" {
```

Comment:
Is it also a requirement that both Variant and Task regexes are provided, if using a regex set? (validTaskSpecifier seems to say it is)

Reply:
It was intended the way you're describing: it wouldn't make sense to only provide one since it would never match anything.
  
I made the front-end/backend validation more consistent: they both check that either a patch alias or a regex set have been set, but not both.
I also made it clearer within the UI. Filling in one section will disable the other and the specifier can't be added until one of {alias, {task_regex, variant_regex}} is provided. I separated them visually as well.

![Screen Shot 2020-11-19 at 12 55 26 PM](https://user-images.githubusercontent.com/17027265/99704546-89b05e80-2a66-11eb-9db4-e26b548c6cf4.png)